subsys/bluetooth/controller/hci/hci.c

```diff
@@ -26,21 +26,28 @@
 
 #include "util/util.h"
 #include "util/memq.h"
+
 #include "hal/ecb.h"
 #include "hal/ccm.h"
+
 #include "ll_sw/pdu.h"
+
 #include "ll_sw/lll.h"
+#include "lll/lll_adv_types.h"
 #include "ll_sw/lll_adv.h"
+#include "lll/lll_adv_pdu.h"
```

Comment:
By past experience, I use files like `*_def.h`, `*_type.h`, `*_func.h` when designing modules. This gives the flexibility to arrange them in groups of defines, types, and functions. And under these groups of includes, considering dependencies, vendor overrides seamlessly merge in. Thoughts are welcome.
 
```suggestion
#include "vendor/lll_adv_type.h" /* redirected to a similar file in vendor (nordic, openisa) folder */
#include "ll_sw/lll_adv_type.h" /* common types */

#include "vendor/lll_adv.h" /* vendor defined function definitions, redirected to vendor folder file */
#include "ll_sw/lll_adv.h" /* common reusable function definitions */
```

Discussions in this PR can be endless. Clearly, currently the upstream code including the vendor `lll_adv_internal.h` is not desired. If the current PR is acceptable solution (may be create a separate GH issue to improve the file organization subsequently), I would like to go ahead with my BIS support being upstreamed.

My 40 odd BIS related commits could take couple calendar weeks with such related discussions popping up. Hope, we can compromise some discussions to avoid too much effort on resolving merge conflicts that arise when up-streaming big list of commits.

Reply:
I think the current PR goes in the right direction. A separate GH issue to discuss how the file organization should look like in the end is a good idea.

Reply:
Created: https://github.com/zephyrproject-rtos/zephyr/issues/32665